packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -0,0 +1,1277 @@
+        } else if (proxyEnd > scrollEnd && position.pixels < position.maxScrollExtent) {
+          final double overDrag = max(proxyEnd - scrollEnd, overDragMax);
+          newOffset = min(position.maxScrollExtent, position.pixels + step * overDrag / overDragCoef);
+        }
+      }
+
+      if (newOffset != null && (newOffset - position.pixels).abs() >= 1.0) {
+        _autoScrolling = true;
+        await position.animateTo(newOffset,
+            duration: duration,
+            curve: Curves.linear
+        );
+        _autoScrolling = false;
+        if (_dragItem != null) {
+          _dragUpdateItems();
+          _autoScroll();
+        }
+      }
+    }
+  }
+
+  SliverChildDelegate _createDelegate() {
+    return SliverChildBuilderDelegate(_itemBuilder, childCount: widget.itemCount + (_reorderingDrag ? 1 : 0));
+  }
+
+  Widget _itemBuilder(BuildContext context, int index) {
+    if (_dragInfo != null && index >= widget.itemCount) {
+      switch (_scrollDirection) {
+        case Axis.horizontal:
+          return SizedBox(width: _dragInfo!.itemExtent);
+        case Axis.vertical:
+          return SizedBox(height: _dragInfo!.itemExtent);
+      }
+    }
+    final Widget child = widget.itemBuilder(context, index);
+    assert(child.key != null, 'All list items must have a key');
+    final OverlayState overlay = Overlay.of(context)!;
+    return _ReorderableItem(
+      key: _ReorderableItemGlobalKey(child.key!, index, this),
+      index: index,
+      child: child,
+      capturedThemes: InheritedTheme.capture(from: context, to: overlay.context),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    assert(debugCheckHasOverlay(context));
+    return SliverList(
+      delegate: _createDelegate(),
+    );
+  }
+}
+
+class _ReorderableItem extends StatefulWidget {
+  const _ReorderableItem({
+    required Key key,
+    required this.index,
+    required this.child,
+    required this.capturedThemes,
+  }) : super(key: key);
+
+  final int index;
+  final Widget child;
+  final CapturedThemes capturedThemes;
+
+  @override
+  _ReorderableItemState createState() => _ReorderableItemState();
+}
+
+class _ReorderableItemState extends State<_ReorderableItem> {
+  late SliverReorderableListState _listState;
+
+  Offset _startOffset = Offset.zero;
+  Offset _targetOffset = Offset.zero;
+  AnimationController? _offsetAnimation;
+
+  Key get key => widget.key!;
+  int get index => widget.index;
+
+  bool get dragging => _dragging;
+  set dragging(bool dragging) {
+    if (mounted) {
+      setState(() {
+        _dragging = dragging;
+      });
+    }
+  }
+  bool _dragging = false;
+
+  @override
+  void initState() {
+    _listState = SliverReorderableList.of(context);
+    _listState._registerItem(this);
+    super.initState();
+  }
+
+  @override
+  void dispose() {
+    _offsetAnimation?.dispose();
+    super.dispose();
```

Comment:
Don't we need to `_listState._unregisterItem(this)` here? If the item was (somehow) explicitly disposed, then it wouldn't have been deactivated first...

Reply:
Ah, my bad. I thought it had to go through deactivate before dispose. Good catch.